Examples/Io/HepMC3/src/HepMC3Event.cpp

```diff
@@ -11,6 +11,82 @@
 #include "ActsExamples/Io/HepMC3/HepMC3Particle.hpp"
 #include "ActsExamples/Io/HepMC3/HepMC3Vertex.hpp"
 
+namespace {
+/// @brief Converts an SimParticle into HepMC3::GenParticle
+/// @note The conversion ignores HepMC status codes
+/// @param actsParticle Acts particle that will be converted
+/// @return converted particle
+HepMC3::GenParticlePtr actsParticleToGen(
+    std::shared_ptr<ActsExamples::SimParticle> actsParticle) {
+  // Extract momentum and energy from Acts particle for HepMC3::FourVector
+  const auto mom4 = actsParticle->momentum4();
+  const HepMC3::FourVector vec(mom4[0], mom4[1], mom4[2], mom4[3]);
+  // Create HepMC3::GenParticle
+  HepMC3::GenParticle genParticle(vec, actsParticle->pdg());
+  genParticle.set_generated_mass(actsParticle->mass());
+
+  return std::shared_ptr<HepMC3::GenParticle>(&genParticle);
+}
+
+/// @brief Converts an Acts vertex to a HepMC3::GenVertexPtr
+/// @note The conversion ignores HepMC status codes
+/// @param actsVertex Acts vertex that will be converted
+/// @return Converted Acts vertex to HepMC3::GenVertexPtr
+HepMC3::GenVertexPtr createGenVertex(
+    const std::shared_ptr<ActsExamples::SimVertex>& actsVertex) {
+  const HepMC3::FourVector vec(
+      actsVertex->position4[0], actsVertex->position4[1],
+      actsVertex->position4[2], actsVertex->position4[3]);
+
+  // Create vertex
+  HepMC3::GenVertex genVertex(vec);
+
+  // Store incoming particles
+  for (auto& particle : actsVertex->incoming) {
+    HepMC3::GenParticlePtr genParticle = actsParticleToGen(
+        std::make_shared<ActsExamples::SimParticle>(particle));
+    genVertex.add_particle_in(genParticle);
+  }
+  // Store outgoing particles
+  for (auto& particle : actsVertex->outgoing) {
+    HepMC3::GenParticlePtr genParticle = actsParticleToGen(
+        std::make_shared<ActsExamples::SimParticle>(particle));
+    genVertex.add_particle_out(genParticle);
+  }
+  return std::shared_ptr<HepMC3::GenVertex>(&genVertex);
+}
+
+/// @brief Compares an Acts vertex with a HepMC3::GenVertex
+/// @note An Acts vertex does not store a barcode. Therefore the content of
+/// both vertices is compared. The position, time and number of incoming and
+/// outgoing particles will be compared. Since a second vertex could exist in
+/// the record with identical informations (although unlikely), this
+/// comparison could lead to false positive results. On the other hand, a
+/// numerical deviation of the parameters could lead to a false negative.
+/// @param actsVertex Acts vertex
+/// @param genVertex HepMC3::GenVertex
+/// @return boolean result if both vertices are identical
+bool compareVertices(const std::shared_ptr<ActsExamples::SimVertex>& actsVertex,
```

Comment:
Not really important, but you could implement as a comparison operator, right?

Reply:
That would mean adding a comparison operator for types external to your library, right? I think having an explicit comparison function here is easier to understand.

Reply:
ok